models/constraints.py

```diff
@@ -0,0 +1,114 @@
+from zineb.exceptions import ConstraintError
+
+class BaseConstraint:
```

Comment:
Complete refactoring of the way constraints work on the model and also removal of the `CheckConstraint` class